### ISSUE TITLE
Module tags not being handled correctly

### ISSUE THREAD

**Issue by probins** [ModuleLoader/es6-module-loader#70]
There are actually 3 problems here, which can easily be demonstrated by making small changes to the demo.

The current `<module>` in the demo is anonymous and inline. According to the spec, an anonymous `<module>` is the equivalent of loader.module(), which executes code. This is not added to the registry, and so can never be imported or referred to again. It could contain exports, but in practice this makes no sense as they can never be referenced. Fine, the current polyfill handles this correctly.

However - **problem 1** - if you change this module to external, with `src="..."` the code is not executed, and because it is anonymous can never be executed because it can't be referenced.

Now add a name to the `<module>`. According to the spec, this is the equivalent of `define()`, which should be added to the registry but _not executed_. The polyfill is handling this correctly when the module is external, but - **problem 2** - is wrongly executing the code if it is inline.

Having now got this named module into the registry, the first `get()` of that module should execute it. As the spec puts it (p30 of the pdf file):
"If the module is in the registry but has never been evaluated, first synchronously evaluate the bodies of the module and any dependencies that have not evaluated yet."
**Problem 3** is that this isn't happening. Although the external named module is correctly added to the registry, there is currently no way of executing it.

As https://gist.github.com/dherman/7568080 puts it:
"[import] is essentially a convenience method for executing and extracting a module after loading it. That is, `l.import(x)` is roughly equivalent to `l.load(x).then(() => Promise.resolve(l.get(x)))`"

in other words, the execution of code in `import()` is part of `get()` not part of `load()`, which is similar to `define()`.

To me, this all highlights the need for expanding the demo to different use cases, not only to show how it works but also to make sure it does in fact work :-)
